x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/IdentifierBuilder.java

```diff
@@ -71,6 +75,9 @@ private static void validateIndexPattern(String indexPattern, EsqlBaseParser.Ind
         String[] indices = indexPattern.split(",");
         boolean hasExclusion = false;
         for (String index : indices) {
+            if (isRemoteIndexName(index)) { // skip the validation if there is remote cluster
```

Comment:
This check is a bit baffling unless you dig in the grammar to understand that when quoting an index pattern, this can actually contain a remote index pattern, which is accepted downstream. I guess a comment (somewhere) would be nice to the next reader, but can also be merged as is without it.